src/status_im/chat/handlers/commands.cljs

```diff
@@ -38,11 +40,14 @@
                 params   {:parameters params
                           :context    (generate-context db chat-id to)}
                 callback #(let [result (get-in % [:result :returned])
-                                result (if (:markup result)
+                                result' (if (:markup result)
                                          (update result :markup cu/generate-hiccup)
-                                         result)]
-                            (dispatch [:set-in [:message-data data-type message-id] result])
-                            (when on-requested (on-requested result)))]
+                                         result)] 
+                            (dispatch [:set-in [:message-data data-type message-id] result'])
+                            (when (= :preview data-type)
```

Comment:
Instead of adding some code specific to `:preview` call here we could pass `on-requested` handler which will then call `messages/update`. We can define  `on-requested` in `:request-command-preview` handler.